src/backend/base/langflow/components/data/Gmail.py

```diff
@@ -0,0 +1,196 @@
+import base64
+import re
+import json
+from typing import Any, Iterator, List
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from langflow.custom import Component
+from langflow.inputs import MessageTextInput
+from langflow.io import SecretStrInput
+from langflow.template import Output
+from langflow.schema import Data
+from langchain_community.chat_loaders.gmail import GMailLoader
+from langchain_core.chat_sessions import ChatSession
+from langchain_core.messages import HumanMessage
+from json.decoder import JSONDecodeError
+from google.auth.exceptions import RefreshError
+
+
+class GmailLoaderComponent(Component):
+    display_name = "Gmail Loader"
+    description = "Loads emails from Gmail using provided credentials."
+    icon = "Google"
+
+    inputs = [
+        SecretStrInput(
+            name="json_string",
+            display_name="JSON String of the Service Account Token",
+            info="JSON string containing OAuth 2.0 access token information for service account access",
+            required=True,
+            value=str("""{
+                "account": "",
+                "client_id": "",
+                "client_secret": "",
+                "expiry": "",
+                "refresh_token": "",
+                "scopes": [
+                    "https://www.googleapis.com/auth/gmail.readonly",
+                ],
+                "token": "",
+                "token_uri": "https://oauth2.googleapis.com/token",
+                "universe_domain": "googleapis.com"
+            }"""),
+        ),
+        MessageTextInput(
+            name="label_ids",
+            display_name="Label IDs",
+            info="Comma-separated list of label IDs to filter emails.",
+            required=True,
+            value="INBOX,SENT,UNREAD,IMPORTANT",
+        ),
+        MessageTextInput(
+            name="max_results",
+            display_name="Max Results",
+            info="Maximum number of emails to load.",
+            required=True,
+            value="10",
+        ),
+    ]
+
+    outputs = [
+        Output(display_name="Loaded Emails", name="emails", method="load_emails"),
+    ]
+
+    def load_emails(self) -> Data:
+        class CustomGMailLoader(GMailLoader):
```

Comment:
Why do you need to create a custom class from it?

Reply:
The Class Gmail loader support only for Label ID "SENT" from emails.

I tweaked the support functions and added cleaning functions.
Now, it support the labels "SENT", "INBOX","UNREAD" and "IMPORTANT". Also the message contnet is cleaned (Extra chareacters removed) using the cleaning function.